File: src/main/java/org/cloudfoundry/credhub/auth/OAuthSignatureException.java
package org.cloudfoundry.credhub.auth;

import org.springframework.security.oauth2.common.exceptions.ClientAuthenticationException;

public class OAuthSignatureException extends ClientAuthenticationException {
  public OAuthSignatureException(String msg) {
    super(msg);
  }

  @Override
  public int getHttpErrorCode() {
    return 401;
  }

  @Override
  public String getOAuth2ErrorCode() {
    return "invalid_token";
  }
}

File: components/auth/src/main/kotlin/org/cloudfoundry/credhub/auth/OAuthSignatureException.kt
package org.cloudfoundry.credhub.auth

import org.springframework.security.oauth2.common.exceptions.ClientAuthenticationException

class OAuthSignatureException(
    msg: String,
) : ClientAuthenticationException(msg) {
    override fun getHttpErrorCode(): Int = 401

    override fun getOAuth2ErrorCode(): String = "invalid_token"
}
